### PR TITLE
Good IN2.72 code in IN1.17 not valid, create no RP

### PR DESCRIPTION
**Pull request by cragun47**
Signed-off-by: Brian Cragun <cragun@us.ibm.com>

Gap in mapping methods allowed a good IN2.72 code to be accepted as valid in IN1.17 where it is not valid.  This was creating RelatedPersons, when they should not be.

The bulk of this is separating the IN1.17 mapping content and DataValueResolvers from the IN2.72 mapping content.   This doubles all the related conditional logic in the Coverage, RelatedPerson, and Patient .yml files.

### CODE REVIEW

**Review by LisaWellman — APPROVED** [review]
Looks good, minor suggestions

File: src/main/resources/hl7/resource/Patient.yml [hunk 1]
```diff
@@ -104,15 +137,32 @@ identifier_5:
    constants:
       system: http://terminology.hl7.org/CodeSystem/v2-0203
 
-# only include when the subscriber is self
+# Only include when the subscriber is self
```
[the remark this answers is not in the document]
Comment (cragun47) [reply src/main/resources/hl7/resource/Patient.yml:140]: Done

File: src/main/resources/hl7/resource/Patient.yml [hunk 2]
```diff
@@ -104,15 +137,32 @@ identifier_5:
...
+    valueIn: IN2.61 | IN1.36
+    subscriberValue: String, IN1.17.1  # subscriber relationship
+    # No systemCX set for this identifier
+  constants:
+    system: "http://terminology.hl7.org/CodeSystem/v2-0203"
+    code: "MB"
+    display: "Member Number"
+
+# Only include when the subscriber is self
+# Give IN1.17 priority (even urecognized value) over IN2.72
```
[the remark this answers is not in the document]
Comment (cragun47) [reply src/main/resources/hl7/resource/Patient.yml:156]: Fixed.  Wording changed completely.

File: src/main/resources/hl7/resource/RelatedPerson.yml [hunk 3]
```diff
@@ -52,14 +52,25 @@ patient:
    generateList: true
    specs: $Patient
 
-relationship:
+relationship_1a:
```
[the remark this answers is not in the document]
Comment (cragun47) [reply src/main/resources/hl7/resource/RelatedPerson.yml:55]: Fixed

File: src/main/resources/hl7/resource/Patient.yml [hunk 4]
```diff
@@ -19,50 +19,83 @@ identifier_1:
...
-   generateList: true
-   expressionType: resource
-   vars:
-      valueIn: String, PID.19 | IN2.2 # subscriber SSN
-      subscriberIsSelf: SUBSCRIBER_IS_SELF, IN1.17 | IN2.72  # subscriber relationship
-   constants:
-      system: http://terminology.hl7.org/CodeSystem/v2-0203
-      code: SS
-      display: Social Security number
+  condition: $valueIn NOT_NULL && $valueIN117 NULL && $valueIN272 NULL
```
[the remark this answers is not in the document]
Comment (cragun47) [reply src/main/resources/hl7/resource/Patient.yml:56]: Correct.  There is no complex logic, which results in extra conditions and verbosity.

**Pull request merged**